src/Microsoft.HttpRepl.IntegrationTests/Commands/PutCommandTests.cs

```diff
@@ -0,0 +1,86 @@
+        {
+            await VerifyErrorMessage(commandText: "PUT",
+                                     baseAddress: null,
+                                     path: null,
+                                     expectedErrorMessage: Resources.Strings.Error_NoBasePath);
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_OnlyBaseAddressWithInlineContent_VerifyResponse()
+        {
+            await VerifyResponse(commandText: "PUT --content \"Test Put Body\"",
+                                 baseAddress: _config.BaseAddress,
+                                 path: null,
+                                 expectedResponseLines: 5,
+                                 expectedResponseContent: "This is a test response from a PUT: \"Test Put Body\"");
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_MultiPartRouteWithInlineContent_VerifyResponse()
+        {
+            await VerifyResponse(commandText: "PUT --content \"Test Put Body\"",
+                                 baseAddress: _config.BaseAddress,
+                                 path: "this/is/a/test/route",
+                                 expectedResponseLines: 5,
+                                 expectedResponseContent: "This is a test response from a PUT: \"Test Put Body\"");
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_MultiPartRouteWithNoBodyRequired_VerifyResponse()
+        {
+            await VerifyResponse(commandText: "PUT --no-body",
+                                 baseAddress: _config.BaseAddress,
+                                 path: "no/body/required",
+                                 expectedResponseLines: 5,
+                                 expectedResponseContent: "This is a test response from a PUT: \"\"");
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_MultiPartRouteWithBodyFromFile_VerifyResponse()
+        {
+            await VerifyResponse(commandText: $"PUT --file \"{nameof(PutCommandTests)}-{nameof(ExecuteAsync_MultiPartRouteWithBodyFromFile_VerifyResponse)}.txt\"",
+                                 baseAddress: _config.BaseAddress,
+                                 path: "this/is/a/test/route",
+                                 expectedResponseLines: 5,
+                                 expectedResponseContent: "This is a test response from a PUT: \"Test Put Body From File\"");
+        }
+    }
+
+    public class PutCommandsConfig : SampleApiServerConfig
+    {
+        public PutCommandsConfig()
+        {
+            Port = SampleApiServerPorts.PutCommandTests;
+            Routes.Add(new DynamicSampleApiServerRoute("PUT", "", RespondWithBody));
+            Routes.Add(new DynamicSampleApiServerRoute("PUT", "this/is/a/test/route", RespondWithBody));
+            Routes.Add(new DynamicSampleApiServerRoute("PUT", "no/body/required", RespondWithBody));
+        }
+
+        private async Task RespondWithBody(HttpContext context)
```

Comment:
This looks pretty much same in every test. Is there a way to not duplicate it?

Reply:
It's the same for now between PUT/POST/PATCH, but I assume they'll diverge as we add more tests for the specific types of calls. If not, I'll come back and clean that up.